models.py

```diff
@@ -2077,3 +2077,245 @@ class FullAdminAccess(Base):
+    )
+    project_id = Column(
+        UUID(as_uuid=True),
+        ForeignKey(f"{Tablenames.PROJECT.value}.id", ondelete="CASCADE"),
+        index=True,
+        nullable=True,
+    )
+    created_by = Column(
+        UUID(as_uuid=True),
+        ForeignKey(f"{Tablenames.USER.value}.id", ondelete="SET NULL"),
+        index=False,
+    )
+    created_at = Column(DateTime, default=sql.func.now())
+    updated_by = Column(
+        UUID(as_uuid=True),
+        ForeignKey(f"{Tablenames.USER.value}.id", ondelete="SET NULL"),
+        index=False,
+    )
+    updated_at = Column(DateTime, onupdate=sql.func.now())
+    started_at = Column(DateTime)
+    finished_at = Column(DateTime)
+    name = Column(String)
+    description = Column(String)
+    tokenizer = Column(String)
+    state = Column(String)  # of type enums.CognitionMarkdownFileState.*.value
+    type = Column(String)  # of type enums.CognitionIntegrationType.*.value
+    config = Column(JSON)
+    """JSON object that contains the configuration for the integration type.
+    Examples:
+        - For a webhook integration, it might contain the URL and headers.
+        - For an API integration, it might contain the API key and endpoint.
+        - For a database integration, it might contain the connection string and credentials.
+
+    """
+
+    llm_config = Column(JSON)
+    error_message = Column(String)
+    is_synced = Column(Boolean, nullable=True)
+    last_synced_at = Column(DateTime)
+    delta_criteria = Column(JSON)
+
+
+class CognitionIntegrationAccess(Base):
+    __tablename__ = Tablenames.INTEGRATION_ACCESS.value
+    __table_args__ = {"schema": "cognition"}
+    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
+    created_by = Column(
+        UUID(as_uuid=True),
+        ForeignKey(f"{Tablenames.USER.value}.id", ondelete="SET NULL"),
+        index=False,
+    )
+    created_at = Column(DateTime, default=sql.func.now())
+    organization_id = Column(
+        UUID(as_uuid=True),
+        ForeignKey(f"{Tablenames.ORGANIZATION.value}.id", ondelete="CASCADE"),
+        index=True,
+    )
+    integration_types = Column(
+        ARRAY(String)
+    )  # of type enums.CognitionIntegrationType.*.value
+
+
+class IntegrationGithubFile(Base):
+    __tablename__ = Tablenames.INTEGRATION_GITHUB_FILE.value
+    __table_args__ = (
+        UniqueConstraint(
+            "integration_id",
+            "running_id",
+            "source",
+            name=f"unique_{__tablename__}_source",
+        ),
+        {"schema": "integration"},
+    )
+    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
+    created_by = Column(
+        UUID(as_uuid=True),
+        ForeignKey(f"{Tablenames.USER.value}.id", ondelete="SET NULL"),
+        index=False,
+    )
+    updated_by = Column(
+        UUID(as_uuid=True),
+        ForeignKey(f"{Tablenames.USER.value}.id", ondelete="SET NULL"),
+        index=False,
+        nullable=True,
```

Comment:
see above & nullable= True is afaik default (otherwise the cascade wouldn't work)